File: src/main/java/ru/usu/data/structure/rope/RopeNodeFactory.java
package ru.usu.data.structure.rope;
public
class RopeNodeFactory
{
    public static RopeNode createNode()
    {
        return new RopeNode();
    }

    /**
     * @return leaf node with string as value
     */
    public static RopeNode createLeafNode(String str)
    {
        RopeNode node = createNode();

        node.influence = str.length();
        node.value = str;

        return node;
    }

    /**
     * @return rope node childs left and right
     */
    public static RopeNode createParentNode(RopeNode left, RopeNode right)
    {
        RopeNode node = createNode();

        node.influence = RopeHelper.getInfluence(left) + RopeHelper.getInfluence(right);
        node.left = left;
        node.right = right;
        node.deep = RopeHelper.getIncDeep(node);
        return node;
    }
}

File: src/main/java/ru/usu/data/structure/rope/RopeNodeFactory.kt
package ru.usu.data.structure.rope

object RopeNodeFactory {
    fun createNode(): RopeNode {
        return RopeNode()
    }

    /**
     * @return leaf node with string as value
     */
    fun createLeafNode(str: String): RopeNode {
        val node = createNode()

        node.influence = str.length
        node.value = str

        return node
    }

    /**
     * @return rope node childs left and right
     */
    fun createParentNode(left: RopeNode?, right: RopeNode?): RopeNode {
        val node = createNode()

        node.influence = RopeHelper.getInfluence(left) + RopeHelper.getInfluence(right)
        node.left = left
        node.right = right
        node.deep = RopeHelper.getIncDeep(node)
        return node
    }
}
